### PR TITLE
Site Settings: Replace wp-admin comments link with the Calypso equivalent

### PR DESCRIPTION
**Pull request by Copons**
Updates the comment moderation queue link in `/settings/discussion/` to point to Calypso instead of wp-admin.

screen shot 2018-01-11 at 18 09 51

### CODE REVIEW

File: client/my-sites/site-settings/form-discussion.jsx [hunk 1]
```diff
@@ -476,13 +477,7 @@ class SiteSettingsFormDiscussion extends Component {
...
 						{
 							components: {
-								link: (
-									<a
-										href={ fields.admin_url + 'edit-comments.php?comment_status=moderated' }
-										target="_blank"
-										rel="noopener noreferrer"
-									/>
-								),
+								link: <a href={ `/comments/all/${ siteSlug }` } />,
```
Comment (rodrigoi) [inline client/my-sites/site-settings/form-discussion.jsx]: `comment_status=moderated` show the _Pending_ comments.

**Comment by rodrigoi** [comment]
found another occurence of `edit-comments.php` here:

https://github.com/Automattic/wp-calypso/blob/dfa5c2da47196682013753be3f8ae854d076193e/client/state/stats/lists/utils.js#L579

maybe it's' worth taking care on this PR 😎

**Comment by Copons** [comment]
> found another occurence of `edit-comments.php` here:
>
> maybe it's' worth taking care on this PR 😎 

Nice catch!

Though: nope, we can't take care of that (yet?).
That link is used in `/stats/insights` and opens the wp-admin comments list filtered by user.
Something that we don't have in Calypso (yet?).

screen shot 2018-01-15 at 12 28 07

---

On the other hand, by switching that stats box view, we get this:

screen shot 2018-01-15 at 12 29 57

The link is built here and opens the post stats.

https://github.com/Automattic/wp-calypso/blob/dfa5c2da47196682013753be3f8ae854d076193e/client/state/stats/lists/utils.js#L597

I'm wondering if it makes sense compared to the "by authors" view links that open the comments moderation.

In other words: it might make more sense to link to the post view in comments moderation.
Though, I don't have a strong opinion on this. 🤔

**Comment by Copons** [comment]
> This would make more sense to me too.

I'd say we can tackle it in another PR, if we consider it worthwhile.
